semgrep-core/src/parsing/Test_parsing.ml

```diff
@@ -29,10 +29,67 @@ let logger = Logging.get_logger [ __MODULE__ ]
 (* Helpers *)
 (*****************************************************************************)
 
+module SMap = Map.Make (String)
+module IMap = Map.Make (Int)
+
+(* Global store, because I don't want to pass it through a bunch of func calls.
+ *)
+let store = ref []
+
+(* This function just processes the exception backtrace to find the first instance where a function
+   that is not called "todo" was called.
+   It then records the line number and name of the function, adding it to a global store.
+*)
+let process_exn () =
+  let rec process (lines : string list) =
+    match lines with
+    | [] -> "failure"
+    | line :: rest -> (
+        (* Sue me. I don't want to write a real lexer and parser. *)
```

Comment:
I usually use a regexp.
if s =~ "Called from \\([^ ]+\\) in [^ ]+ line \\([0-9]+\\)"
then
  let (funcname, line) = Common.matched2 s in
 ...